pkg/instrumentation/auto/annotation.go

```diff
@@ -26,9 +25,10 @@ const (
 
 // +kubebuilder:rbac:groups="",resources=namespaces,verbs=list;watch;update
 
-// AnnotationMutators has an AnnotationMutator resource name
+// AnnotationMutators contains functions that can be used to mutate annotations
+// on all supported objects based on the configured mutators.
 type AnnotationMutators struct {
-	client              client.Client
+	clientWriter        client.Writer
```

Comment:
Curious whats the difference between the two ? Or is it coz we have separated out the reader functionality to ```client.Reader``` ?

Reply:
The `client.Client` implements both the `client.Writer` (Create, Update, Patch, Delete) and `client.Reader` (Get, List) along with a bit more. Separating it out makes it easier to mock for testing and also makes it clear what each is being used for.